File: backend/src/models/Preset.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const PresetSchema = new Schema(
  {
    userId: {
      type: Schema.Types.ObjectId,
      ref: 'User',
      required: true,
      index: true,
    },
    name: {
      type: String,
      required: true,
      trim: true,
    },
    calories: {
      type: Number,
      required: true,
      min: 0,
    },
    proteinGrams: {
      type: Number,
      required: true,
      min: 0,
    },
    carbohydratesGrams: {
      type: Number,
      required: true,
      min: 0,
    },
    fatGrams: {
      type: Number,
      required: true,
      min: 0,
    },
    amount: {
      type: Number,
      default: 1,
      min: 0.1,
    },
  },
  {
    timestamps: true,
  }
);

// Search by name
PresetSchema.index({ userId: 1, name: 'text' });

module.exports = mongoose.model('Preset', PresetSchema);

File: mobile/lib/core/models/preset.dart
class Preset {
  final String? id;
  final String userId;
  final String name;
  final double calories;
  final double proteinGrams;
  final double carbohydratesGrams;
  final double fatGrams;
  final double amount;

  Preset({
    this.id,
    required this.userId,
    required this.name,
    required this.calories,
    required this.proteinGrams,
    required this.carbohydratesGrams,
    required this.fatGrams,
    this.amount = 1.0,
  });

  factory Preset.fromJson(Map<String, dynamic> json) {
    return Preset(
      id: json['_id'],
      userId: json['userId'],
      name: json['name'],
      calories: (json['calories'] as num).toDouble(),
      proteinGrams: (json['proteinGrams'] as num).toDouble(),
      carbohydratesGrams: (json['carbohydratesGrams'] as num).toDouble(),
      fatGrams: (json['fatGrams'] as num).toDouble(),
      amount: (json['amount'] as num?)?.toDouble() ?? 1.0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'calories': calories,
      'proteinGrams': proteinGrams,
      'carbohydratesGrams': carbohydratesGrams,
      'fatGrams': fatGrams,
      'amount': amount,
    };
  }
}
